### PR TITLE
Send events

### PR DESCRIPTION
**Pull request by steve-dickinson**

### CODE REVIEW

File: app/config/mq-config.js [hunk 1]
```diff
@@ -12,6 +12,12 @@ const mqSchema = joi.object({
     address: joi.string().default('payment'),
     username: joi.string(),
     password: joi.string()
+  },
+  eventTopic: {
+    name: joi.string().default('ffc-pay-event'),
```
Comment (johnwatson484) [inline app/config/mq-config.js:17]: Doesn't need a name.  I'd remove it to simplify unless the npm package needs it.

File: app/config/mq-config.js [hunk 2]
```diff
@@ -12,6 +12,12 @@ const mqSchema = joi.object({
...
+    address: joi.string().default('payment'),
```
Comment (johnwatson484) [inline app/config/mq-config.js:18]: I'd remove the default.   This causes errors when local env vars are setup.  It prevents failing fast.

File: app/event/raise-event.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const { PublishEvent } = require('ffc-pay-event-publisher')
+const { eventTopic } = require('../config/mq-config')
+
+const raiseEvent = async (event, status = 'success', error = '') => {
```
Comment (johnwatson484) [inline app/event/raise-event.js:4]: `error` parameter isn't used so can be removed.

File: app/event/send-batch-capture-event.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const { v4: uuidv4 } = require('uuid')
+const raiseEvent = require('../event/raise-event')
+
+const sendBatchCaptureEvent = async (correlation) => {
+  const event = {
+    id: uuidv4(),
+    name: 'batch-processing-capture-event',
```
Comment (johnwatson484) [inline app/event/send-batch-capture-event.js:7]: Do the event names need `-event` on the end?  It seems redundant

File: app/event/send-batch-capture-event.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    type: 'info',
+    message: 'Correlation for Payment request within Siti payment file',
```
Comment (johnwatson484) [inline app/event/send-batch-capture-event.js:9]: Nitpick: `Payment` doesn't need a capital

**Review by johnwatson484 — COMMENTED** [review]
Added a few suggestions for improvement, nothing blocking.

Not sure if there's a way we can make the raising error functions more generic so we don't need a specific one for each event.

The README and event API spec need updating to show the new correlation Id

File: app/config/mq-config.js (hunk 2 above)
Comment (simonnineteen90) [reply app/config/mq-config.js:18]: removed

File: app/event/raise-event.js (hunk 3 above)
Comment (simonnineteen90) [reply app/event/raise-event.js:4]: removed

File: app/config/mq-config.js (hunk 1 above)
Comment (simonnineteen90) [reply app/config/mq-config.js:17]: removed

**Comment by simonnineteen90** [comment]
made suggested changes and updated the readme and async api

**Review by johnwatson484 — APPROVED** [review]

**Pull request merged**